blockchain/system/registry.go

```diff
@@ -65,10 +65,12 @@ func AllocRegistry(init *params.RegistryConfig) map[common.Hash]common.Hash {
 	// slot[1]: string[] names;
 	// - names.length @ 1
 	// - names[i] @ Hash(1) + i
-	storage[lpad32(1)] = lpad32(len(names)) // names.length
+	storage[lpad32(1)] = lpad32(len(names))
 	for i, name := range names {
-		nameSlot := calcArraySlot(1, 1, i, 0)
-		storage[nameSlot] = encodeShortString(name) // names[i]
+		nameSlot := calcArraySlot(1, 1, i, 0) // Hash(1) + 1*i + 0
+		for k, v := range allocDynamicData(nameSlot, []byte(name)) {
```

Comment:
this looks like a loop, but `allocDynamicData` always returns a map of size 1.
How about changing `allocDynamicData()` to return `(Hash slot, Hash val)`